src/modules/px4iofirmware/mixer.cpp

```diff
@@ -287,19 +295,29 @@ mixer_tick(void)
 		/* mix */
 		mixed = mixer_mix_threadsafe(&outputs[0], &r_mixer_limits);
 
-		/* the pwm limit call takes care of out of band errors */
-		pwm_limit_calc(should_arm, should_arm_nothrottle, mixed, r_setup_pwm_reverse, r_page_servo_disarmed,
-			       r_page_servo_control_min, r_page_servo_control_max, outputs, r_page_servos, &pwm_limit);
+		/* if mixer_mix_threadsafe returns zero, it did nothing */
+		if (mixed != 0) {
```

Comment:
Like I said - I am not familiar with this part of the system. So I am deferring to the experts.

My assumption is the original author (OA) had an intent. Just as @kd0aij does. 

@kd0aij Do I understand, correctly, that your intent is to only update the Oneshots on valid Mix?

As far as the OA -  Could this be perhaps a form of "lazy initialization" or have some side effect I do not understand?  

@kd0aij  If your intent was not to fire the update. Then proposed change does that and does not change the original intent.  

To me that is the safest approach.  

@kd0aij  If it does not matter if up_pwm_force_update is called on a mad mix, then whole if  can come out.